### PR TITLE
Refactor/hotkey swap

### PR DESCRIPTION
**Pull request by distributedstatemachine**
Devnet companion: https://github.com/opentensor/subtensor/pull/567

## Description

This PR :

refactors the hot key swap function to make it more testable
adds unit tests

## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please describe): refactor and testing 

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

File: pallets/subtensor/tests/registration.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ fn test_registration_subscribe_ok_dispatch_info_ok() {
         assert_eq!(
             call.get_dispatch_info(),
             DispatchInfo {
-                weight: frame_support::weights::Weight::from_parts(192_000_000, 0),
+                weight: frame_support::weights::Weight::from_parts(2_992_000_000, 0),
```
Comment (sam0x17) [inline pallets/subtensor/tests/registration.rs:41]: that's quite a big weight change, just flagging but probably fine

**Review by sam0x17 — DISMISSED** [review]

File: pallets/subtensor/src/swap.rs [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_stake(old_hotkey: &T::AccountId, new_hotkey: &T::AccountId, weight: &mut Weight) {
+        let mut writes = 0;
+        let stakes: Vec<(T::AccountId, u64)> = Stake::<T>::iter_prefix(old_hotkey).collect();
+        let stake_count = stakes.len() as u32;
+        for (coldkey, stake_amount) in stakes {
+            Stake::<T>::insert(new_hotkey, &coldkey, stake_amount);
+            let _ = Stake::<T>::clear_prefix(old_hotkey, stake_count, None);
```
Comment (open-junius) [inline pallets/subtensor/src/swap.rs]: can put it out of loop, just execute once.

Comment (distributedstatemachine) [reply pallets/subtensor/src/swap.rs]: done

**Review by sam0x17 — APPROVED** [review]

**Review by open-junius — APPROVED** [review]

File: pallets/subtensor/src/lib.rs [hunk 3]
```diff
@@ -262,6 +266,11 @@ pub mod pallet {
     pub fn TotalSupply<T: Config>() -> u64 {
         21_000_000_000_000_000 // Rao => 21_000_000 Tao
     }
+    /// Hotkey swap cost.
+    #[pallet::type_value]
+    pub fn HotkeySwapCost<T: Config>() -> u64 {
```
Comment (JohnReedV) [inline pallets/subtensor/src/lib.rs]: Not sure this is actually being used? Seems like it is using `InitialHotkeySwapCost` as the default instead.

Comment (distributedstatemachine) [reply pallets/subtensor/src/lib.rs]: good catch! will fix

Comment (distributedstatemachine) [reply pallets/subtensor/src/lib.rs]: done

**Review by JohnReedV — APPROVED** [review]

**Pull request merged**